library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -183,4 +203,35 @@ private void deleteBatchesForIds(List<Long> batchIdsToDelete, Collection<FileDow
     public Cursor retrieveFor(BatchQuery query) {
         return resolver.query(downloadsUriProvider.getBatchesUri(), null, query.getSelection(), query.getSelectionArguments(), query.getSortOrder());
     }
+
+    private static class StatusCountMap {
+
+        private final SparseArrayCompat<Integer> statusCounts = new SparseArrayCompat<>(PRIORITISED_STATUSES_SIZE);
+
+        public boolean hasNoItemsWithStatuses(List<Integer> excludedStatuses) {
+            boolean hasOtherItems = false;
+            for (int status : excludedStatuses) {
+                boolean hasItemsForStatus = hasCountFor(status);
+                if (hasItemsForStatus) {
+                    hasOtherItems = true;
+                    break;
+                }
+            }
+            return hasOtherItems;
+        }
+
+        public boolean hasCountFor(int status) {
+            return statusCounts.get(status, 0) > 0;
```

Comment:
maybe you can avoid some unboxing by using the `compareTo` method in the `Integer` class
http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#compareTo(java.lang.Integer)

http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/lang/Integer.java#Integer.compareTo%28java.lang.Integer%29